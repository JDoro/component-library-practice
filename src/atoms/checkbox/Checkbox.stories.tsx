import type {Meta, StoryObj} from '@storybook/react';

import {useArgs} from '@storybook/preview-api';
import {Checkbox} from './Checkbox.component.tsx';

const meta = {
  title: 'Atoms/Checkbox',
  component: Checkbox,
  tags: ['autodocs'],
  args: {
    label: 'Option Name',
  },
  render: (args) => {
    const [{ value, onChange }, updateArgs] = useArgs();
    return (
      <Checkbox
        {...args}
        value={value}
        onChange={(value) => {
          onChange?.(value);
          updateArgs({ value });
        }}
      />
    );
  }
} satisfies Meta<typeof Checkbox>;

export default meta;
type Story = StoryObj<typeof meta>;

export const All: Story = {};

