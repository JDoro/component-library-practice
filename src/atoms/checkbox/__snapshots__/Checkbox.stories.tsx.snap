// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Checkbox All smoke-test 2`] = `
<div class="flex items-center">
  <input id="checkbox"
         type="checkbox"
         class="h-5 w-5 rounded-md border-[1px] border-secondary"
         style
  >
  <label class="ml-2"
         for="checkbox"
  >
    Option Name
  </label>
</div>
`;
